syntax_and_semantics/operators.md

```diff
@@ -21,101 +45,271 @@ struct Vector2
+| Binary AND      | `&`                                                  |
+| Binary OR/XOR   | `\|`, `^`                                            |
+| Equality        | `==`, `!=`, `=~`, `!~`, `===`                        |
+| Comparison      | `<`, `<=`, `>`, `>=`, `<=>`                          |
+| Logical AND     | `&&`                                                 |
+| Logical OR      | `\|\|`                                               |
+| Range           | `..`, `...`                                          |
+| Conditional     | `?:`                                                 |
+| Assignment      | `=`, `[]=`, `+=`, `&+=`, `-=`, `&-=`, `*=`, `&*=`, `/=`, `//=`, `%=`, `\|=`, `&=`, `^=`, `**=`, `<<=`, `>>=`, `\|\|=`, `&&=` |
 
-  def []=(index, value)
-    # ...
-  end
-end
+## Operators explained
 
-array = MyArray.new
+### Arithmetic operators
 
-array[1]       # invokes the first method
-array[1, 2, 3] # invokes the second method
-array[1] = 2   # invokes the third method
+#### Unary
 
-array.[](1)       # invokes the first method
-array.[](1, 2, 3) # invokes the second method
-array.[]=(1, 2)   # invokes the third method
-```
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `+`  | positive | `+1` | yes |
+| `&+` | wrapping positive | `&+1` | yes |
+| `-`  | negative | `-1` | yes |
+| `&-` | wrapping negative | `&-1` | yes |
+
+#### Multiplicative
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `**` | exponentiation | `1 ** 2` | yes |
+| `&**` | wrapping exponentiation | `1 &** 2` | yes |
+| `*` | multiplication | `1 * 2` | yes |
+| `&*` | wrapping multiplication | `1 &* 2` | yes |
+| `/` | division | `1 / 2` | yes |
+| `//` | floor division | `1 // 2` | yes |
+| `%` | modulus | `1 % 2` | yes |
+
+#### Additive
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `+` | addition | `1 + 2` | yes |
+| `&+` | wrapping addition | `1 &+ 2` | yes |
+| `-` | subtraction | `1 - 2` | yes |
+| `&-` | wrapping subtraction | `1 &- 2` | yes |
+
+### Other unary operators
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `!` | inversion | `!true` | no |
+| `~` | binary complement | `~1` | yes |
+
+### Shifts
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `<<` | shift left, append | `1 << 2`, `STDOUT << "foo"` | yes |
+| `>>` | shift right | `1 >> 2` | yes |
+
+### Binary
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `&` | binary AND | `1 & 2` | yes |
+| `\|` | binary OR | `1 \| 2` | yes |
+| `^` | binary XOR | `1 ^ 2` | yes |
+
+### Equality
+
+Three base operators test equality:
+
+* `==`: Checks whether the values of the operands are equal
+* `=~`: Checks whether the value of the first operand matches the value of the
+second operand with pattern matching.
+* `===`: Checks whether the values of the operands are equal in [case equality](case.html).
```

Comment:
Maybe it's worth noticing that `===` is not symmetric. `t === u` calls `T#===(u : U)` and means `u` belongs or matches `t`.

Reply:
No operator in Crystal is intrinsically commutative. Even `A#==(B)` can be different from `B#==(A)`, depending on the implementation. That should be avoided because it's confusing, but it is possible.
But yeah, I'll try to improve the wording for `===`. It's excplicitly *not* desgined as commutative.